features/cellular/TARGET_GENERIC_MODEM/generic_modem_driver/ReferenceCellularInterface.h

```diff
@@ -0,0 +1,271 @@
+     *  an SMS or GPS extension.
+     *
+     *  @param sim_pin     PIN for the SIM card
+     *  @param apn         optionally, access point name
+     *  @param uname       optionally, Username
+     *  @param pwd         optionally, password
+     *  @return            NSAPI_ERROR_OK on success, or negative error code on failure
+     */
+    virtual nsapi_error_t connect(const char *sim_pin, const char *apn = 0,
+                                  const char *uname = 0, const char *pwd = 0);
+
+
+    /** Attempt to connect to the Cellular network
+     *
+     *  Brings up the network interface. Connects to the Cellular Radio
+     *  network and then brings up the underlying network stack to be used
+     *  by the cellular modem over PPP interface.
+     *
+     *  If the SIM requires a PIN, and it is not set/invalid, NSAPI_ERROR_AUTH_ERROR is returned.
+     *  For APN setup, default behaviour is to use 'internet' as APN string and assuming no authentication
+     *  is required, i.e., username and password are no set. Optionally, a database lookup can be requested
+     *  by turning on the APN database lookup feature. In order to do so, add 'MBED_REF_CELL_DRV_APN_LOOKUP'
+     *  in your mbed_app.json. APN database is by no means exhaustive. It contains a short list of some public
+     *  APNs with publicly available usernames and passwords (if required) in some particular countries only.
+     *  Lookup is done using IMSI (International mobile subscriber identifier).
+     *  Please note that even if 'MBED_REF_CELL_DRV_APN_LOOKUP' config option is set, 'set_credentials()' api still
+     *  gets the precedence. If the aforementioned API is not used and the config option is set but no match is found in
+     *  the lookup table then the driver tries to resort to default APN settings.
+     *
+     *  Preferred method is to setup APN using 'set_credentials()' API.
+
+     *  @return         0 on success, negative error code on failure
+     */
+    virtual nsapi_error_t connect();
+
+    /** Attempt to disconnect from the network
+     *
+     *  Brings down the network interface. Shuts down the PPP interface
+     *  of the underlying network stack. Does not bring down the Radio network
+     *
+     *  @return         0 on success, negative error code on failure
+     */
+    virtual nsapi_error_t disconnect();
+
+    /** Adds or removes a SIM facility lock
+     *
+     * Can be used to enable or disable SIM pin check at device startup.
+     * This API sets up flags for the driver which would either enable or disable
+     * SIM pin checking depending upon the user provided argument while establishing
+     * connection. It doesn't do anything immediately other than setting up flags.
+     *
+     * @param check      can be set to true if the SIM pin check is supposed to be enabled
+     *                   and vice versa.
+     */
+    void check_sim_pin(bool check);
```

Comment:
Nit: As a verb, this function name makes me think this function actively does something, but your comment makes it clear what the function is actually doing.

Thoughts on just naming this `ReferenceCellularInterface::enabled_sim_pin_check`?